op-challenger2/game/fault/test/claim_builder.go

```diff
@@ -161,20 +161,33 @@ func (c *ClaimBuilder) claim(pos types.Position, opts ...ClaimOpt) types.Claim {
 		},
 		AttackBranch: cfg.branch,
 	}
-	if cfg.claimant != (common.Address{}) {
-		claim.Claimant = cfg.claimant
-	}
-	if cfg.value != (common.Hash{}) {
-		claim.Value = cfg.value
-	} else if cfg.invalidValue {
-		claim.Value = c.incorrectClaim(pos)
-	} else {
-		claim.Value = c.CorrectClaimAtPosition(pos)
-		// when nbits is 1, subValues is also filled with claim.Value
-		claim.SubValues = &[]common.Hash{claim.Value}
-	}
 	if cfg.subValues != nil {
 		claim.SubValues = cfg.subValues
+		claim.Value = contracts.SubValuesHash(*claim.SubValues)
+	} else {
+		values := []common.Hash{}
+		if pos.IsRootPosition() {
+			if cfg.invalidValue {
+				values = append(values, c.incorrectClaim(pos))
+			} else {
+				values = append(values, c.CorrectClaimAtPosition(pos))
+			}
+			for i := uint64(0); i < c.MaxAttackBranch()-1; i++ {
+				values = append(values, common.Hash{})
```

Comment:
After rethinking the issue that `len(subValues)` at `Root` and `SplitDepth+Nbits` is 1, I think it's better to handle it in contract. So, there is no need to pad ` common.Hash{}` to `values`.

Same for `GetClaimsAtPosition` and `ExpectAttackV2` func.